ddtrace/tracer.py

```diff
@@ -579,6 +579,8 @@ def _start_span(
                 # We must always mark the span as sampled so it is forwarded to the agent
                 span.sampled = True
 
+        span._executor_id = self.context_provider._executor_id()
```

Comment:
calling out that I want to reread/revisit this

Reply:
discussed off of GitHub, we want to move this to `ContextProvider.activate` instead.

moving to activate means we can verify that the span you are trying to activate belongs to the same execution as the context.

as well, we can ensure that if a span does not have an execution id set, then it will be set to the one where it is activated, not where it was created.

e.g.

```python
span = tracer.start_span(activate=False)

# start a new execution

tracer.context_provider.activate(span)
```